fiftyone/utils/complex_yolo/model.py

```diff
@@ -1,50 +1,225 @@
+        box_corners = self._get_box_corners(x, y, w, l, angle)
+        bbox1 = np.array(box_corners).reshape((-1, 4, 2))
+        bbox1 = self._convert_boxes_to_shapely(bbox1)
+
+        bbox2 = []
+        for i in range(box2.shape[0]):
+            x, y, w, l, im, re = box2[i, :]
+            angle = np.arctan2(im, re)
+            bev_corners = self._get_box_corners(x, y, w, l, angle)
+            bbox2.append(bev_corners)
+
+        bbox2 = self._convert_boxes_to_shapely(np.array(bbox2))
+
+        return self._compute_iou(bbox1[0], bbox2)
+
+    def nms(self, prediction, conf_thresh, nms_thresh):
+        """Removes detections with lower object confidence score than
+        `conf_thresh` and performs non-maximum suppression to further filter
+        detections.
+
+        Returns detections with shape::
+
+            (x, y, w, l, im, re, object_conf, class_score, class_pred)
+        """
+        output = [None for _ in range(len(prediction))]
+        for image_i, image_pred in enumerate(prediction):
+            # Filter out confidence scores below threshold
+            image_pred = image_pred[image_pred[:, 6] >= conf_thresh]
+
+            # If none are remaining => process next image
+            if not image_pred.size(0):
+                continue
+
+            # Object confidence times class confidence
+            score = image_pred[:, 6] * image_pred[:, 7:].max(1)[0]
+
+            # Sort by it
+            image_pred = image_pred[(-score).argsort()]
+
+            class_confs, class_preds = image_pred[:, 7:].max(1, keepdim=True)
+            detections = torch.cat(
+                (
+                    image_pred[:, :7].float(),
+                    class_confs.float(),
+                    class_preds.float(),
+                ),
+                1,
+            )
+
+            keep_boxes = []
+            while detections.size(0):
+                large_overlap = (
+                    self._rotated_bbox_iou_polygon(
+                        detections[0, :6], detections[:, :6]
+                    )
+                    > nms_thresh
+                )
+                large_overlap = torch.from_numpy(large_overlap)
+                label_match = detections[0, -1] == detections[:, -1]
+
+                # Indices of boxes with lower confidence scores, large IOUs and
+                # matching labels
+                invalid = large_overlap & label_match
+                weights = detections[invalid, 6:7]
+
+                # Merge overlapping bboxes by order of confidence
+                detections[0, :6] = (weights * detections[invalid, :6]).sum(
+                    0
+                ) / weights.sum()
+                keep_boxes += [detections[0]]
+                detections = detections[~invalid]
+
+            if keep_boxes:
+                output[image_i] = torch.stack(keep_boxes)
+
+        return output
+
+    def compute_height(
+        self, predictions, classes, class_heights=None, class_zs=None
```

Comment:
I updated this method's signature so that `class*` are explicitly passed in as arguments. Previously some private attributes of the instance were being secretly set by the user of the model, which is confusing for someone trying to figure out how the model functions in general and what this method does in particular.